mpeg2/src/temi.rs

```diff
@@ -0,0 +1,282 @@
+            17
+        } else {
+            5
+        };
+        if self.ntp_timestamp.is_some() {
+            len += 8;
+        }
+        if self.ptp_timestamp.is_some() {
+            len += 10;
+        }
+
+        len += if self.has_timecode == TimestampLength::ThirtyTwoBits {
+            7
+        } else if self.has_timecode == TimestampLength::SixtyFourBits {
+            12
+        } else {
+            0
+        };
+        len
+    }
+
+    pub fn encode<W: Write>(&self, mut w: W) -> Result<usize, EncodeError> {
+        let mut buf = vec![0u8; self.encode_len()];
+        buf[0] = AF_DESCR_TAG_TIMELINE;
+        buf[2] = (self.has_timestamp as u8) << 6;
+
+        if self.force_reload {
+            buf[2] |= 0b0000_0010;
+        }
+        if self.paused {
+            buf[2] |= 0b0000_0001;
+        }
+        if self.discontinuity {
+            buf[3] = 0b1000_0000;
+        }
+        buf[4] = self.timeline_id;
+
+        let mut ret = 5;
+        if self.has_timestamp != TimestampLength::None {
+            BigEndian::write_u32(&mut buf[5..9], self.timescale);
+            if self.has_timestamp == TimestampLength::ThirtyTwoBits {
+                BigEndian::write_u32(&mut buf[9..13], self.media_timestamp as u32);
+                ret += 8;
+            } else if self.has_timestamp == TimestampLength::SixtyFourBits {
+                BigEndian::write_u64(&mut buf[9..17], self.media_timestamp);
+                ret += 12;
+            }
+        }
+
+        if let Some(ntp_ts) = self.ntp_timestamp {
+            buf[2] |= 0b0010_0000;
+            BigEndian::write_u64(&mut buf[ret..(ret + 8)], ntp_ts);
+            ret += 8;
+        }
+        if let Some(ptp_ts) = self.ptp_timestamp {
+            buf[2] |= 0b0001_0000;
+            BigEndian::write_u16(&mut buf[ret..(ret + 2)], (ptp_ts >> 64) as u16);
+            ret += 2;
+            BigEndian::write_u64(&mut buf[ret..(ret + 8)], ptp_ts as u64);
+            ret += 8;
+        }
+
+        if self.has_timecode != TimestampLength::None {
+            buf[2] |= (self.has_timecode as u8) << 2;
+            if self.drop {
+                buf[ret] = 0b1000_0000;
+            }
+            buf[ret] |= (self.frames_per_tc_seconds >> 8) as u8 & 0b0111_1111;
+            buf[ret + 1] = self.frames_per_tc_seconds as _;
+            buf[ret + 2] = (self.duration >> 8) as u8;
+            buf[ret + 3] = self.duration as _;
+
+            ret += 4;
+            if self.has_timecode == TimestampLength::ThirtyTwoBits {
+                for i in 0..=2 {
+                    buf[ret + i] = (self.time_code >> ((2 - i) * 8)) as u8;
+                }
+                ret += 3;
+            } else {
+                BigEndian::write_u64(&mut buf[ret..(ret + 8)], self.time_code);
+                ret += 8;
+            }
+        }
+
+        buf[1] = ret as u8 - 2;
+
+        w.write_all(&buf)?;
+        Ok(buf.len())
```

Comment:
We have a variable named `ret`. We should either return it or name it something else like "offset".

```suggestion
        Ok(ret)
```

Reply:
they're equal, should simplify return `ret`.